src/Compilers/Test/Core/Platform/Custom/MetadataSignatureHelper.cs

```diff
@@ -323,7 +323,7 @@ public static StringBuilder AppendMethodImplAttributes(StringBuilder sb, MethodI
 
             sb.Append(codeType);
             sb.Append(' ');
-            sb.Append((attributes & MethodImplAttributes.Unmanaged) == MethodImplAttributes.Unmanaged ? "unmanaged" : "managed");
+            sb.Append((attributes & MethodImplAttributes.ManagedMask) == MethodImplAttributes.Unmanaged ? "unmanaged" : "managed");
```

Comment:
The current pattern doesn't actually appear to be a problem? Whether something is managed or not is just 1 bit.

Reply:
right, same here. I don't have a particularly strong opinion other than not spreading a bad pattern.